compiler/rustc_target/src/asm/powerpc.rs

```diff
@@ -51,7 +51,11 @@ impl PowerPCInlineAsmRegClass {
                 }
             }
             Self::freg => types! { _: F32, F64; },
-            Self::vreg => &[],
+            // FIXME: vsx also supports integers and floats: https://github.com/rust-lang/rust/pull/131551#discussion_r1797651773
```

Comment:
This is mostly an LLVM limitation. We *can* work around it with special support code in our LLVM lowering, but it's optional.

Reply:
I did a little more looking into the issue here:

- Floats(f32/f64) are allowed for v constraint (`in/out(vreg)`), but not for concrete register names (`in/out("v[0-31]"`). (LLVM error)
- Integers are not allowed in both v constraint and concrete register names. (LLVM error)

So, I adjusted our LLVM lowering to allow floats in concrete `v[0-31]` register names, and I left it unsupported for integers for now.